scripts/build.ps1

```diff
@@ -179,26 +179,21 @@ param (
 Set-StrictMode -Version 'Latest'
 $PSDefaultParameterValues['*:ErrorAction'] = 'Stop'
 
-if ("" -eq $Arch) {
-    if ($IsMacOS) {
-        $RunningArch = uname -m
-        if ("x86_64" -eq $RunningArch) {
-            $IsTranslated = sysctl -in sysctl.proc_translated
-            if ($IsTranslated) {
-                $Arch = "arm64"
-            } else {
-                $Arch = "x64"
-            }
-        } elseif ("arm64" -eq $RunningArch) {
-            $Arch = "arm64"
-        } else {
-            Write-Error "Unknown architecture"
-        }
-    } else {
-        $Arch = "x64"
-    }
+$BuildConfig = @{
+    Platform = $Platform
+    Tls = $Tls
+    Arch = $Arch
+    ExtraArtifactDir = $ExtraArtifactDir
+    Config = $Config
 }
 
+$BuildConfig = & (Join-Path $PSScriptRoot helpers.ps1) -BuildConfig $BuildConfig
```

Comment:
Can you have a better name than `helpers.ps1`?

Reply:
Also, instead of using the pattern you have so far, I think I'd prefer to pass all the options directly into the script, use positional arguments (so you don't need to do `-Platform <..> -Tls ...` every time, and then just use `$BuildConfig.*` directly, and not create the temporary variables like `$Tls`.

Reply:
The goal was to be able to add more options to this. Something like `ScriptHelpers.ps1` better?

Reply:
Why not make this specific to build config and just make this `Get-BuildConfig.ps1` or something like that?

Reply:
That idea doesn't work Nick, as I need to return multiple values, which requires objects. Also, this script I also want to move other helper functions used in multiple places into.

Reply:
I didn't say to change how/what you return from the script. So I don't quite understand what doesn't work. What other helpers do you want to move? I just don't like the idea of a generic "helper" script because you have no clue what's inside it from the name.